proto/catalog.proto

```diff
@@ -226,6 +236,19 @@ message Subscription {
   SubscriptionState subscription_state = 19;
 }
 
+message ConnectionParams {
```

Comment:
This is kind of a *weak-typed* implementation, as the properties are kept as a `map<string,string>` here.

On the contrary, we may optionally make it *strong-typed* by defining connection's fields as a `message` (such as `message KafkaConnection`), and meanwhile, the corresponding struct (such as `struct KafkaConnectionInner`) would be substituted.

I tend to prefer the latter one, but I am not sure how to handle secrets. Comments are welcomed.